src/utils/__tests__/formatNumber.test.ts

```diff
@@ -1,151 +1,262 @@
-    expect(formatAmount(999_999_000_000_000)).toEqual('> 999T')
-  })
+      expect(formatAmount(1_100_000_000_000)).toEqual('1.1T')
+      expect(formatAmount(1_234_567_898_765.123456789)).toEqual('1.235T')
+      expect(formatAmount(100_001_000_000_000)).toEqual('100.001T')
+      expect(formatAmount(999_999_000_000_000)).toEqual('> 999T')
+    })
 
-  it('should use > 999T for numbers above 999,000,000,000,000', () => {
-    expect(formatAmount(999_000_000_000_001)).toEqual('> 999T')
-    expect(formatAmount(999_000_000_000_000.001)).toEqual('> 999T')
-  })
+    it('should use > 999T for numbers above 999,000,000,000,000', () => {
+      expect(formatAmount(999_000_000_000_001)).toEqual('> 999T')
+      expect(formatAmount(999_000_000_000_000.001)).toEqual('> 999T')
+    })
+
+    it('should use < 0.00001 for amounts smaller then 0.00001', () => {
+      expect(formatAmount(0.00001)).toEqual('0.00001')
+      expect(formatAmount(0.000014)).toEqual('0.00001')
+      expect(formatAmount(0.000015)).toEqual('0.00002')
+      expect(formatAmount(0.000001)).toEqual('< 0.00001')
+      expect(formatAmount(0.000009)).toEqual('< 0.00001')
+    })
+
+    it('should use < -0.00001 or < +0.00001 when the Eucledian distance of the amount is smaller than 0.00001', () => {
+      // to keep the '+' sign the amount shall be passed as a string
+      expect(formatAmount('+0.000001')).toEqual('< +0.00001')
+      expect(formatAmount('+0.000009')).toEqual('< +0.00001')
 
-  it('should use < 0.00001 for amounts smaller then 0.00001', () => {
-    expect(formatAmount(0.00001)).toEqual('0.00001')
-    expect(formatAmount(0.000014)).toEqual('0.00001')
-    expect(formatAmount(0.000015)).toEqual('0.00002')
-    expect(formatAmount(0.000001)).toEqual('< 0.00001')
-    expect(formatAmount(0.000009)).toEqual('< 0.00001')
+      // negative numbers will keep the sign either way
+      expect(formatAmount(-0.000001)).toEqual('< -0.00001')
+      expect(formatAmount(-0.000009)).toEqual('< -0.00001')
+      expect(formatAmount('-0.000001')).toEqual('< -0.00001')
+      expect(formatAmount('-0.000009')).toEqual('< -0.00001')
+    })
   })
 
-  it('should use < -0.00001 or < +0.00001 when the Eucledian distance of the amount is smaller than 0.00001', () => {
-    // to keep the '+' sign the amount shall be passed as a string
-    expect(formatAmount('+0.000001')).toEqual('< +0.00001')
-    expect(formatAmount('+0.000009')).toEqual('< +0.00001')
+  describe('formatCurrency', () => {
+    it('returns the correct number of decimals', () => {
+      const amount1 = 1
+
+      expect(formatCurrency(amount1, 'JPY')).toBe('1 JPY')
+      expect(formatCurrency(amount1, 'IQD')).toBe('1 IQD')
+      expect(formatCurrency(amount1, 'USD')).toBe('1.00 USD')
+      expect(formatCurrency(amount1, 'EUR')).toBe('1.00 EUR')
+      expect(formatCurrency(amount1, 'GBP')).toBe('1.00 GBP')
+      expect(formatCurrency(amount1, 'BHD')).toBe('1.000 BHD')
+
+      const amount2 = '0.123456789'
```

Comment:
The `amount2` rounds always down for the tested currencies. I would add a `amount3 = 1.7777` (per example) to test if the rounding up is working as expected as well.